mffpy/header_block/header_block.py

```diff
@@ -105,10 +76,18 @@ def __new__(cls,
         num_samples : sample count per channel in the block
         sampling_rate : sampling_rate per channel in the block
         header_size : byte size of the header (computed if None)
+        optional : tuple containing optional header fields
         """
-        header_size = header_size or cls.compute_byte_size(num_channels)
+        if header_size:
+            computed_header_size = cls.compute_byte_size(
+                num_channels, optional)
+            assert header_size == computed_header_size, f"""
+            inconsistent header {header_size} != {computed_header_size}"""
```

Comment:
Would raising a `ValueError` make more sense here instead of an `AssertionError`?